MovieQuiz/Services/StatisticService.swift

```diff
@@ -0,0 +1,86 @@
+
+import Foundation
+
+protocol StatisticService {
+    func store(correct count: Int, total amount: Int)
+    var totalAccuracy: Double { get }
+    var gamesCount: Int { get }
+    var bestGame: GameRecord { get }
+}
+
+
+class StatisticServiceImplementation: StatisticService {
+
+    private enum Keys: String {
+        case correct, total, bestGame, gamesCount
+    }
+
+    func store(correct count: Int, total amount: Int) {
+
+        let currentGameResults = GameRecord(correct: count, total: amount, date: Date())
+
+        if self.bestGame < currentGameResults {
+            self.bestGame = currentGameResults
+        }
+
+        let correctStored = userDefaults.integer(forKey: Keys.correct.rawValue)
+        userDefaults.set(correctStored + count, forKey: Keys.correct.rawValue)
+
+        let totalStored = userDefaults.integer(forKey: Keys.total.rawValue)
+        userDefaults.set(totalStored + amount, forKey: Keys.total.rawValue)
+
+        userDefaults.set(gamesCount + 1, forKey: Keys.gamesCount.rawValue)
+    }
+
+    var totalAccuracy: Double {
+        let correctStored = userDefaults.double(forKey: Keys.correct.rawValue)
+        let totalStored = userDefaults.double(forKey: Keys.total.rawValue)
+        return (correctStored / totalStored) * 100
+    }
+
+    var gamesCount: Int {
+        get {
+            userDefaults.integer(forKey: Keys.gamesCount.rawValue)
+            }
```

Comment:
Удобно использовать встроенный форматер, в простых случаях будет работать хорошо: выделить код и зажать`control+i`
```suggestion
        }
```